### PR TITLE
Circular dependency validation

### PR DESCRIPTION
**Pull request by mitresh95**
| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Automated tests included? | N/A
| Related user documentation PR URL | N/A
| Related developer documentation PR URL | N/A
| Issues addressed (#s or URLs) | N/A
| BC breaks? | N/A
| Deprecations? | N/A

[//]: # ( Required: )
#### Description:
Segment A can exclude Segment B and Segment B exclude Segment A. We need a validation for when the form is saved to check any segments included in a filter to see if those segment refer back to the one being saved. If so, throw a validation error.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Add Segment A as a member in filters of Segment B and do the same with Segment B in filters of Segment A.
2. The operation will be completed without any validation errors.

#### Steps to test this PR:
1. Apply PR.
2. Perform the same steps as to reproduce the issue and this time it will prevent the operation from getting completed and show the validation error message.

### CODE REVIEW

**Comment by mleffler** [comment]
Tested once and code reviewed

File: app/bundles/CoreBundle/Translations/en_US/validators.ini [hunk 1]
```diff
@@ -1,5 +1,6 @@
 mautic.core.ab_test.winner_criteria.not_blank="Select a winning criteria."
 mautic.core.email.required="A valid email is required."
+mautic.core.segment.circular_dependency_exists="Update cannot be completed as there are one or more circular dependencies in this segment. Please remove the dependency for the segment to function."
```
Comment (Woeler) [inline app/bundles/CoreBundle/Translations/en_US/validators.ini]: I would suggest changing this to something along the lines of: `Update cannot be completed as the current filters are conflicting with the filters of another segment. Make sure your segment filters are not creating impossible or contradictory conditions (e.g. Segment A must be a member of Segment B while Segment B must be a member of Segment A).`

File: app/bundles/LeadBundle/Views/List/form.html.php [hunk 2]
```diff
@@ -119,6 +119,11 @@ class="segment-filter <?php echo $icon; ?>">
                             <div class="clearfix"></div>
                         </div>
                         <div class="selected-filters" id="leadlist_filters">
+                            <?php if ($filterErrors): ?>
+                                <div class="alert alert-danger">
```
Comment (Woeler) [inline app/bundles/LeadBundle/Views/List/form.html.php]: Add the css class `has-error` before `alert` and it should style like any other red error box in Mautic.

Comment (kuzmany) [reply app/bundles/LeadBundle/Views/List/form.html.php]: Thanks. I've add it.

**Comment by kuzmany** [comment]
@escopecz  there is 3 positive reviews. 
Hope we can merge. 
You can decide about translation suggestion from us  👍

**Pull request merged**